draft/docstrings.jl

```diff
@@ -0,0 +1,255 @@
+    "MPolyRingElem" "MPolyRingElem" "RingElem" "A multivariate polynomial" "polynomial_ring(R :x :y)"
+]
+
+for (base, _, _) in eachrow(base_rings), (name, _, _, _, _) in eachrow(constructed_rings)
+    eval(:(struct $(Symbol(base*name)) end))
+end
+for (_, super, _, _, _) in eachrow(constructed_rings)
+    eval(:(struct $(Symbol(super)) end))
+end
+module Generic
+for (_, super, _, _, _) in eachrow(Main.constructed_rings)
+    eval(:(struct $(Symbol(super)) end))
+end
+end
+
+for (base, ring, latex) in eachrow(base_rings), (name, super, _, description, see) in eachrow(constructed_rings)
+    eval(:(
+        @doc Markdown.parse("""
+            $($(base)*$(name)) <: $($(super)){$($(base)*$(String(ring)))Elem}
+
+        $($(description)) over $('$'*$(latex)*'$'). See [`$($(see))`](@ref).
+        """)
+        $(Symbol(base*name))
+   ))
+end
+
+for (_, name, _, description, see) in eachrow(constructed_rings)
+    eval(:(
+    @doc Markdown.parse("""
+        Generic.$($(name)){T} <: $($(name)){T}
+
+    $($(description)) with coefficients of type `T`. Generic implementation for arbitrary rings.
+    See [`$($(see))`](@ref).
+    """)
+    Generic.$(Symbol(name))
+   ))
+end
+
+for (_, name, super, description, see) in eachrow(constructed_rings)
+    eval(:(
+    @doc Markdown.parse("""
+        $($(name)){T} <: $($(super)){T}
+
+    $($(description)) with coefficients of type `T`. See [`$($(see))`](@ref).
+    """)
+    $(Symbol(super))
+   ))
+end
+
+
+
+# == Ergänzungen einiger Dokumentationen ==
+
+@doc md"""
+    matrix(R::Ring, arr::AbstractMatrix)
+
+Construct the matrix over $R$ with entries as in `arr`.
+
+
+# Examples
+
+```jldoctest
+julia> matrix(ZZ, [0 1 2; 3 4 5]
+[0   1   2]
+[3   4   5]
+```
+"""
+matrix(::Ring, ::AbstractMatrix)
+
+@doc md"""
+    matrix(R::Ring, r::Int, c::Int, arr::AbstractVector)
+
+Construct the $r\times c$ matrix over $R$, where the entries are taken row-wise from `arr`.
+
+See also [`matrix_space`](@ref).
+
+# Examples
+
+```jldoctest
+julia> matrix(ZZ, 2, 3, [0,1,2,3,4,5])
+[0   1   2]
+[3   4   5]
+```
+"""
+matrix(::Ring, ::Int, ::Int, ::AbstractVector)
+
+@doc md"""
+    matrix_space(R::NCRing, r::Int, c::Int; cached::Bool = true)
+
+Return parent object corresponding to the space of $r\times c$ matrices over the ring $R$.
+
+If `cached == true`, the returned parent object is cached, so that it can be returned by future calls to the constructor with the same dimensions and base ring.
+
+# Examples
+
+```jldoctest
+julia> ZZ2x3 = MatrixSpace(ZZ, 2, 3)
+Matrix Space of 2 rows and 3 columns over Integer Ring
+
+julia> ZZ2x3([0,1,2,3,4,5])
+[0   1   2]
+[3   4   5]
+
+julia> ZZ2x3([0 1 2; 3 4 5])
+[0   1   2]
+[3   4   5]
+
+julia> ZZ2x3(7)
+[7   0   0]
+[0   7   0]
+```
+"""
+matrix_space
+
+@doc md"""
+    polynomial_ring(R, :x)
+
+Return the polynomial ring over $R$ and the variable $x$.
+Any ring works, and $x$ can be given as any `AbstractString` or `Char`.
```

Comment:
or `Symbol` (as your example shows)

Reply:
Oh right, `Symbol`s are no `AbstractString`s. Not completely intuitive for me.
Anyway, the docstring currently lies, when it claims, that $x$ can be given as `AbstractString`, `Char`, or `Symbol`. I made up #1405 to let that docstring eventually become true 😄